app/models/infra_conversion_job.rb

```diff
@@ -408,6 +415,54 @@ def poll_shutdown_vm_complete
     abort_conversion(error.message, 'error')
   end
 
+  def optimize_vm_placement
+    update_migration_task_progress(:on_entry)
+
+    if !migration_task.warm_migration? && target_vm.vendor == 'vmware' && Settings.transformation.optimize_vmware_vm_placement
+      eligible_vmware_hosts = []
+      target_vm.ems_cluster.hosts.each do |host|
+        next if host.is_tagged_with?('transformation_eligible_for_disk_transfer/false', :ns => 'managed')
+        next if (target_vm.storages & host.storages) != target_vm.storages
```

Comment:
I'm curious about this logic here. Actually, some comments on this method in general explaining the logic at a high level for the method overall would be great.